wp-content/plugins/aircraft/aircraft.js

```diff
@@ -0,0 +1,26 @@
+( function( wp ) {
+
+    const aircraftShortcode = function ( props ) {
+        return wp.element.createElement(
+            wp.editor.RichTextToolbarButton, {
+                icon: 'editor-code',
+                title: 'Aircraft Shortcode',
+                onClick: function() {
+                    props.onChange( wp.richText.toggleFormat(
```

Comment:
I know this is where I need to call the shortcode to run when a user clicks on the format button. I just don't know how to actually reference that shortcode from this js file.

Reply:
Thanks for digging into this!

I think part of where the confusion lies is that what we're actually trying to do is different than what the toggleFormat method does (that just happened to be what the tutorial was about). I went and looked at the richText package and there is an [insert](https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-rich-text/#insert) method that seems more in line with what it is we're actually trying to do, which is insert a string `[aircraft]` into the post.

There isn't a lot of documentation or examples out there on this, so you might have to dig a little bit to find an existing example or you'll have to figure it out from trial and error. I did find in the Gutenberg codebase the code they use for inserting a [line break](https://github.com/WordPress/gutenberg/blob/186b672ac8bba85ac113158019cf8722d821867c/packages/rich-text/src/insert-line-separator.js), which might be similar enough that you can use it for reference. Here are a few other articles I found that might point you in the right direction:

- https://neliosoftware.com/blog/how-to-add-button-gutenberg-blocks/
- https://gist.github.com/rmorse/1300421889ec7fbb9217899e61ab703d